src/kernel/arch/x86/gdt.zig

```diff
@@ -532,106 +532,106 @@ test "makeEntry alternating bit pattern" {
         .granularity = 0,
     };
 
-    expectEqual(u4(0b0101), @bitCast(u4, alt_flag));
+    expectEqual(@as(u4, 0b0101), @bitCast(u4, alt_flag));
 
-    const actual = makeEntry(u32(0b01010101010101010101010101010101), u20(0b01010101010101010101), alt_access, alt_flag);
+    const actual = makeEntry(0b01010101010101010101010101010101, 0b01010101010101010101, alt_access, alt_flag);
 
-    const expected = u64(0b0101010101010101010101010101010101010101010101010101010101010101);
+    const expected: u64 = 0b0101010101010101010101010101010101010101010101010101010101010101;
     expectEqual(expected, @bitCast(u64, actual));
 }
 
 test "setTssStack" {
     // Pre-testing
-    expectEqual(u32(0), tss.prev_tss);
-    expectEqual(u32(0), tss.esp0);
-    expectEqual(u32(KERNEL_DATA_OFFSET), tss.ss0);
-    expectEqual(u32(0), tss.esp1);
-    expectEqual(u32(0), tss.ss1);
-    expectEqual(u32(0), tss.esp2);
-    expectEqual(u32(0), tss.ss2);
-    expectEqual(u32(0), tss.cr3);
-    expectEqual(u32(0), tss.eip);
-    expectEqual(u32(0), tss.eflags);
-    expectEqual(u32(0), tss.eax);
-    expectEqual(u32(0), tss.ecx);
-    expectEqual(u32(0), tss.edx);
-    expectEqual(u32(0), tss.ebx);
-    expectEqual(u32(0), tss.esp);
-    expectEqual(u32(0), tss.ebp);
-    expectEqual(u32(0), tss.esi);
-    expectEqual(u32(0), tss.edi);
-    expectEqual(u32(0), tss.es);
-    expectEqual(u32(0), tss.cs);
-    expectEqual(u32(0), tss.ss);
-    expectEqual(u32(0), tss.ds);
-    expectEqual(u32(0), tss.fs);
-    expectEqual(u32(0), tss.gs);
-    expectEqual(u32(0), tss.ldtr);
-    expectEqual(u16(0), tss.trap);
-    expectEqual(u16(@sizeOf(TtsEntry)), tss.io_permissions_base_offset);
+    expectEqual(@as(u32, 0), tss.prev_tss);
```

Comment:
I think if you swap the arguments around you may not need to cast

Reply:
True, but then if I e.g. change 0 to 1 then it fails with "expected 0, found 1", but in reality 1 is the expected value.

Reply:
True

Reply:
It would be nice to not have to cast, but I reckon the expectEqual function needs to be changed for that.